packages/bp/src/admin/workspace/collaborators/collaborators-router.ts

```diff
@@ -40,16 +40,19 @@ class CollaboratorsRouter extends CustomAdminRouter {
       '/listAvailableUsers',
       this.needPermissions('read', this.resource),
       this.asyncMiddleware(async (req, res) => {
-        const filterRoles = req.query.roles && req.query.roles.split(',')
-        const allUsers = await this.authService.getAllUsers()
-        const workspaceUsers = await this.workspaceService.getWorkspaceUsers(req.workspace!)
-        const available = _.filter(allUsers, x => !_.find(workspaceUsers, x)) as WorkspaceUser[]
+        if (!req.workspace) {
+          throw new BadRequestError('Workspace must be defined to list available users')
+        }
 
-        return sendSuccess(
-          res,
-          'Retrieved available users',
-          filterRoles ? available.filter(x => filterRoles.includes(x.role)) : available
-        )
+        const filterRoles = (req.query.roles && req.query.roles.split(',')) || []
+        const filterAuthStrategies = (await this.workspaceService.findWorkspace(req.workspace)).authStrategies || []
+
+        const allUsers = await this.workspaceService.getAllWorkspaceUsers()
+
+        const available = _.filter(allUsers, u => u.workspace !== req.workspace)
```

Comment:
You are fetching ALL users of ALL workspaces, then you filter them if the workspace doesn't match ? Why not simply use `this.workspaceService.getWorkspaceUsers(req.workspace)` instead ?

Reply:
Filter works the other way, we're _keeping_ all the users that aren't in the workspace - this is basically the inverse of `getWorkspaceUsers`